### PR TITLE
enhance: add disk quota and max collections into db properties

### PR DESCRIPTION
**Pull request by jaime0815**
issue: #34385

### CODE REVIEW

File: internal/rootcoord/quota_center.go [hunk 1]
```diff
@@ -1356,6 +1402,7 @@ func (q *QuotaCenter) checkDiskQuota() error {
 		}
 	}
 
+	// check disk quota of db, partition and collection level
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:1405]: confusing comment

File: internal/rootcoord/quota_center.go [hunk 2]
```diff
@@ -1326,21 +1351,42 @@ func (q *QuotaCenter) checkDiskQuota() error {
...
 			dbs = append(dbs, dbID)
 		}
 	}
 
+	//  DB properties take precedence over quota configuration for disk quota.
+	for dbID, binlogSize := range dbSizeInfo {
+		db, err := q.meta.GetDatabaseByID(q.ctx, dbID, typeutil.MaxTimestamp)
+		if err == nil {
+			if dbDiskQuotaStr := db.GetProperty(common.DatabaseDiskQuotaKey); dbDiskQuotaStr != "" {
+				if dbDiskQuotaBytes, err := strconv.ParseFloat(dbDiskQuotaStr, 64); err == nil {
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:1380]: Wrap a GetDiskQuota function

File: internal/rootcoord/quota_center.go [hunk 3]
```diff
@@ -1326,21 +1351,42 @@ func (q *QuotaCenter) checkDiskQuota() error {
...
+					dbDiskQuotaMB := dbDiskQuotaBytes * 1024 * 1024
+					checkDiskQuota(dbID, binlogSize, dbDiskQuotaMB)
+				}
+			}
+		} else {
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:1385]: should not be "else"?

File: internal/rootcoord/quota_center.go [hunk 4]
```diff
@@ -1326,21 +1351,42 @@ func (q *QuotaCenter) checkDiskQuota() error {
 			log.Warn("cannot find db id for collection", zap.Int64("collection", collection))
 			continue
 		}
+
+		// skip db that has already denied writing
```
Comment (bigsheeper) [inline internal/rootcoord/quota_center.go:1355]: Pass the denyWritingDBs param is not very elegant.

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**